File: robocode2-server/src/main/java/net/robocode2/model/Score.java
package net.robocode2.model;

import lombok.Builder;
import lombok.Value;

/**
 * Defines a score record to keep track of a bot's score
 * 
 * @author Flemming N. Larsen
 */
@Value
@Builder(toBuilder=true)
public class Score {

	/** Bot ID */
	int botId;

	/** Survival score gained whenever another bot is defeated */
	double survival;

	/** Last survivor score as last survivor in a round */
	double lastSurvivorBonus;

	/** Bullet damage given */
	double bulletDamage;

	/** Bullet kill bonus */
	double bulletKillBonus;

	/** Ram damage given */
	double ramDamage;

	/** Ram kill bonus */
	double ramKillBonus;

	/** Total score */
	public double getTotalScore() {
		return survival + lastSurvivorBonus + bulletDamage + bulletKillBonus + ramDamage + ramKillBonus;
	}

	/** Number of 1st places */
	int firstPlaces;

	/** Number of 2nd places */
	int secondPlaces;

	/** Number of 3rd places */
	int thirdPlaces;
}

File: robocode-tankroyale-server-kotlin/src/main/kotlin/dev/robocode/tankroyale/server/model/Score.kt
package dev.robocode.tankroyale.server.model

/** Defines a score record to keep track of a bot's score. */
data class Score(
    /** Bot ID */
    var botId: Int,

    /** Survival score gained whenever another bot is defeated */
    var survival: Double = 0.0,

    /** Last survivor score as last survivor in a round */
    var lastSurvivorBonus: Double = 0.0,

    /** Bullet damage given */
    var bulletDamage: Double = 0.0,

    /** Bullet kill bonus */
    var bulletKillBonus: Double = 0.0,

    /** Ram damage given */
    var ramDamage: Double = 0.0,

    /** Ram kill bonus */
    var ramKillBonus: Double = 0.0,

    /** Number of 1st places  */
    var firstPlaces: Int = 0,

    /** Number of 2nd places  */
    var secondPlaces: Int = 0,

    /** Number of 3rd places  */
    var thirdPlaces: Int = 0
) {
    /** Total score */
    val totalScore: Double
        get() = survival + lastSurvivorBonus + bulletDamage + bulletKillBonus + ramDamage + ramKillBonus
}